### PR TITLE
Feature  Farscape's get 

### PR DESCRIPTION
**Pull request by jcarres-mdsol**
Add classes to easily get and discover resources, it uses crichton-representors to deserialize them

@fosdev @chad-medi  @cabbott @sheavalentine-mdsol @brutski

### CODE REVIEW

File: lib/farscape/agent/result.rb [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
     class Result < Struct.new(:request, :response)
       extend Forwardable
-      
+
       def_delegators :response, :status, :headers, :body
 
+      def deserialize
+        deserializer = Crichton::Deserializer.create(response.headers['Content-Type'], response.body)
+        deserializer.deserialize
+      rescue Crichton::UnknownFormatError
```
Comment (cabbott) [inline lib/farscape/agent/result.rb]: going to rescue and do nothing?

File: lib/farscape/config_file.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  # This class manages the configuration from the Farscape config file.
+  # This class is totally unrelated to Farscape::Configuration even when the names are similar
+  # TODO: Rename this or that or do something about this.
+  class ConfigFile
+    def configuration
+      default_config = {
+        default_accept: 'application/vnd.hale+json'
+      }
+      default_config.merge!(config_data )
```
Comment (cabbott) [inline lib/farscape/config_file.rb]: nitpick: extraneous space

File: lib/farscape/agent/result.rb (hunk 1 above)
Comment (ghost) [inline lib/farscape/agent/result.rb]: +1.  do something, right?

File: lib/farscape/config_file.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    def configuration
+      default_config = {
+        default_accept: 'application/vnd.hale+json'
+      }
+      default_config.merge!(config_data)
+    end
+
+    private
+    def config_data
+      config_hash = symbolize_keys(YAML::load(IO.read(Dir.glob(CONFIGURATION_FILE_PATH).first.to_s)))
```
Comment (ghost) [inline lib/farscape/config_file.rb]: YAML.load_file with do the IO.read for you, right?

File: lib/farscape/config_file.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    rescue Errno::ENOENT
+      puts "YAML configuration file couldn't be found. Discovery not available."
```
Comment (ghost) [inline lib/farscape/config_file.rb]: maybe give some message about where it expects to find the YAML.

File: lib/farscape/config_file.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    rescue Psych::SyntaxError
+      puts "YAML configuration file contains invalid syntax. Discovery not available."
+    ensure #return is needed here or this method would return nil on exceptions
+      return config_hash || {}
```
Comment (ghost) [inline lib/farscape/config_file.rb:24]: `return` is unnecessary.

Comment (jcarres-mdsol) [inline lib/farscape/config_file.rb:24]: In fact it is. Believe or not without this the method will return nil. Ruby glitch?

File: lib/farscape/config_file.rb (hunk 3 above)
Comment (jcarres-mdsol) [inline lib/farscape/config_file.rb]: load only loads a string

Comment (ghost) [inline lib/farscape/config_file.rb]: No.  YAML.load_file accepts a string representating a path to a file, and then reads the file: http://www.ruby-doc.org/stdlib-1.9.3/libdoc/syck/rdoc/Syck.html

File: lib/farscape/config_file.rb (hunk 5 above)
Comment (ghost) [inline lib/farscape/config_file.rb:24]: Ruby, go home.

File: lib/farscape/config_file.rb (hunk 3 above)
Comment (jcarres-mdsol) [inline lib/farscape/config_file.rb]: you are right, done!

**Comment by jcarres-mdsol** [comment]
@chad-medi @cabbott  
This should be ready!

**Pull request merged**